agent/consul/acl.go

```diff
@@ -416,7 +416,7 @@ func (r *ACLResolver) resolveIdentityFromToken(token string) (structs.ACLIdentit
 	waitForResult := cacheEntry == nil || r.config.ACLDownPolicy != "async-cache"
 	if !waitForResult {
 		// waitForResult being false requires the cacheEntry to not be nil
-		return cacheEntry.Identity, nil
+		return cacheEntry.Identity, cacheEntry.Error
```

Comment:
Since the caller has special handling for `ACLRemoteError`, we need to make sure we wrap the error like we do in L430, otherwise we'll miss the `ACLRemoteError` codepath

```suggestion
		if cacheEntry.Error != nil && !acl.IsErrNotFound(cacheEntry.Error) {
			return cacheEntry.Identity, ACLRemoteError{Err: cacheEntry.Error}
		}
		return cacheEntry.Identity, cacheEntry.Error
```

Reply:
Thanks, I totally missed the part about needing the return the err wrapped in an `ACLRemoteError` - I thought the bug was only that we weren't caching the error at all and returning an ErrNotFound when we shouldn't be